### PR TITLE
Kan 8 users features

### PR DESCRIPTION
**Pull request by GeeA02**

### CODE REVIEW

File: server/src/app/features/users/query-handlers/get-user-details.query.handler.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  constructor(private dependencies: GetUserDetailsQueryHandlerDependencies) {}
+
+  async execute(query: GetUserDetailsQuery): Promise<GetUserDetailsQueryResult> {
+    const { userRepository } = this.dependencies;
+    const { id } = query.payload;
+    const user = await userRepository.findOne({ id });
+    if (user) {
+      const { username, name, surname } = user;
+      return new GetUserDetailsQueryResult({ username, name, surname });
```
Comment (Nynu8) [inline server/src/app/features/users/query-handlers/get-user-details.query.handler.ts:26]: Could replace both lines with `return new GetUserDetailsQueryResult({ ...user })`

File: server/src/app/features/users/query-handlers/get-user-details.query.handler.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+    throw new Error("User does not exist");
```
Comment (Nynu8) [inline server/src/app/features/users/query-handlers/get-user-details.query.handler.ts:28]: Enter plox

File: server/src/app/features/users/query-handlers/get-user-details.query.handler.ts (hunk 1 above)
Comment (Nynu8) [inline server/src/app/features/users/query-handlers/get-user-details.query.handler.ts:27]: Could replace both lines with `return new GetUserDetailsQueryResult({ ...user })`

File: server/src/app/features/users/query-handlers/get-user-details.query.handler.ts (hunk 2 above)
Comment (Nynu8) [inline server/src/app/features/users/query-handlers/get-user-details.query.handler.ts:30]: Enters plox

**Review by Nynu8 — CHANGES_REQUESTED** [review]

File: server/src/app/features/users/query-handlers/get-user-details.query.handler.ts (hunk 2 above)
Comment (GeeA02) [reply server/src/app/features/users/query-handlers/get-user-details.query.handler.ts:30]: done

File: server/src/app/features/users/query-handlers/get-user-details.query.handler.ts (hunk 1 above)
Comment (GeeA02) [reply server/src/app/features/users/query-handlers/get-user-details.query.handler.ts:27]: It was ok

**Pull request merged**